src/mock_display.rs

```diff
@@ -563,12 +633,195 @@ where
+            // a single row of characters.
+            continue;
+        }
+
+        write_row(f, display, bounding_box, y, 0)?;
+        f.write_char('\n')?
+    }
+
+    Ok(())
+}
+
+fn write_row<C>(
+    f: &mut fmt::Formatter<'_>,
+    display: &MockDisplay<C>,
+    bounding_box: &Rectangle,
+    y: i32,
+    column_width: usize,
+) -> fmt::Result
+where
+    C: PixelColor + ColorMapping,
+{
+    for x in bounding_box.columns() {
+        let point_top = Point::new(x, y);
+        let point_bottom = Point::new(x, y + 1);
+
+        // Set foreground color.
+        if bounding_box.contains(point_top) {
+            let color = display
+                .get_pixel(point_top)
+                .map(|c| c.into())
+                .unwrap_or(C::NONE_COLOR);
+
+            write!(f, "\x1b[38;2;{};{};{}m", color.r(), color.g(), color.b())?;
+        } else {
+            write!(f, "\x1b[39m")?;
+        };
+
+        // Set background color.
+        if bounding_box.contains(point_bottom) {
+            let color = display
+                .get_pixel(point_bottom)
+                .map(|c| c.into())
+                .unwrap_or(C::NONE_COLOR);
+
+            write!(f, "\x1b[48;2;{};{};{}m", color.r(), color.g(), color.b())?;
+        } else {
+            write!(f, "\x1b[49m")?;
+        };
+
+        // Write "upper half block" character.
+        f.write_char('\u{2580}')?;
+    }
+
+    // Reset colors.
+    f.write_str("\x1b[0;m")?;
+
+    // Pad output with spaces if column width is larger than the width of the bounding box.
+    for _ in bounding_box.size.width as usize..column_width {
+        f.write_char(' ')?
+    }
+
+    Ok(())
+}
+
+fn write_vertical_border(f: &mut fmt::Formatter<'_>, column_width: usize) -> fmt::Result {
+    write!(
+        f,
+        "+-{:-<width$}-+-{:-<width$}-+-{:-<width$}-+\n",
+        "",
+        "",
+        "",
+        width = column_width
+    )
+}
+
+fn write_header(f: &mut fmt::Formatter<'_>, column_width: usize) -> fmt::Result {
+    write_vertical_border(f, column_width)?;
+    write!(
+        f,
+        "| {:^width$} | {:^width$} | {:^width$} |\n",
+        "display",
+        "expected",
+        "diff",
+        width = column_width
+    )?;
+    write_vertical_border(f, column_width)
+}
+
+impl<C> Display for FancyPanic<'_, C>
+where
+    C: PixelColor + ColorMapping,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let diff = self.display.diff(self.expected);
+
+        let bounding_box_display = self.display.affected_area_origin();
+        let bounding_box_expected = self.expected.affected_area_origin();
+
+        let bounding_box = Rectangle::new(
+            Point::zero(),
+            bounding_box_display
+                .size
+                .component_max(bounding_box_expected.size),
+        );
+
+        f.write_char('\n')?;
+
+        // Output the 3 displays in columns if they are less than 30 pixels wide.
+        if bounding_box.size.width <= 30 {
```

Comment:
Nice! I was going to complain that the boxes might not fit in thinner terminals but here's this `if` statement just sitting there!

Reply:
The current value of 30 pixels will produce an output of up to 110 characters per line. Should we reduce the width further to make sure that it also fits in the default 80 columns?

Reply:
I think 110 is fine. Not many of us are developing e-g on a C64 any more (although your old Intel box is getting close ;))

Reply:
> Not many of us are developing e-g on a C64 any more...

That's why I suggested 80 and not 40 :wink:.

> ...(although your old Intel box is getting close ;))

True, but my ultrawide monitor isn't quite as old as the CPU and can easily display more than 400 characters per row in a relatively large font.